crates/amaru-consensus/src/consensus/headers_tree/headers_tree.rs

```diff
@@ -507,8 +551,43 @@ impl<H: IsHeader + Clone + PartialEq + Eq> HeadersTree<H> {
     }
 
     /// Return the best currently known tip
-    fn best_chain(&self) -> &HeaderHash {
-        &self.best_chain
+    fn best_chain(&self) -> HeaderHash {
+        self.chain_store.get_best_chain_hash()
+    }
+
+    /// Store the best currently known tip
+    fn set_best_chain(&mut self, hash: &HeaderHash) -> Result<(), ConsensusError> {
+        self.chain_store
+            .set_best_chain_hash(hash)
+            .map_err(|e| ConsensusError::SetBestChainHashFailed(*hash, e))?;
+        // update "Me" so that it points to the best chain, as given by the the best chain tip
+        self.peers.insert(Me, self.best_chain_fragment_hashes());
+        Ok(())
+    }
+
+    /// Store the best currently known tip
```

Comment:
Comment might be off or imprecise?

Reply:
It's the same as the previous function. Seems to me these comments do not add anything useful?

Reply:
Yes I fixed the comments. Especially the one for `update_best_chain` since the point is to do an atomic update.